protocol/cmd/dydxprotocold/cmd/root.go

```diff
@@ -85,12 +83,24 @@ func NewRootCmdWithInterceptors(
 	appConfigInterceptor func(string, *DydxAppConfig) (string, *DydxAppConfig),
 	appInterceptor func(app *dydxapp.App) *dydxapp.App,
 ) *cobra.Command {
-	encodingConfig := dydxapp.GetEncodingConfig()
+	tempApp := dydxapp.New(
```

Comment:
why is there a temp app?

Reply:
This is how Cosmos recommends to build the v1 app https://docs.cosmos.network/v0.50/build/migrations/upgrading#autocli
https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-beta.0/simapp/app.go#L636-L655

Effectively you have to build an instance of the app to build another instance of the app again.

It seems as though if we used depinject (aka v2 type app) we wouldn't have to do this.